src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -48,6 +49,25 @@ public static string GetResourceName(ResourceViewModel resource, IEnumerable<Res
     }
 }
 
+public sealed class CommandViewModel
+{
+    public string CommandType { get; }
+    public string DisplayName { get; }
+    public string? ConfirmationMessage { get; }
+    public Value? Parameter { get; }
+
+    public CommandViewModel(string commandType, string displayName, string? confirmationMessage, Value? parameter)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(commandType);
+        ArgumentException.ThrowIfNullOrWhiteSpace(displayName);
+
+        CommandType = commandType;
+        DisplayName = displayName;
```

Comment:
Assuming we would want to localize the displayName?

Reply:
I assume we will want that localized somehow, along with things like custom statuses and whatnot. But I'm not sure what the plan is for that. Will discuss with folks, but I think that can come in a separate PR.

Reply:
We should consider sending user culture to hosting and letting hosting localize